stl/inc/flat_map

```diff
@@ -744,6 +744,24 @@ public:
         return {lower_bound(_X), upper_bound(_X)};
     }
 
+    friend bool operator==(const flat_map& _X, const flat_map& _Y) {
+        return _RANGES equal(_X._Data.keys, _Y._Data.keys) && _RANGES equal(_X._Data.values, _Y._Data.values);
+    }
+
+    friend _STD weak_ordering operator<=>(const flat_map& _X, const flat_map& _Y) {

```

Comment:
> A flat_map meets all of the requirements of a container ([[container.reqmts]](https://eel.is/c++draft/container.reqmts))

This should be `strong_ordering`:
![image](https://github.com/microsoft/STL/assets/60953653/51772829-2951-4d99-b1c5-25c7fd971009)